### ISSUE TITLE
LeViT training and bench on GTSRB dataset 

### ISSUE THREAD

**Issue by thaihoangminhtam** [facebookresearch/LeViT#29]
Hello

I'm trying to use your SOTA LeViT for GTSRB but encounter some problems when testing. The accuracy after testing 12K images in GTSRB was only 13.4%, 347 FPS on 3080Ti. I believe that your model could break any record of my survey and training may be the main cause. I have tried levit.py and levit_c.py to load the model with only arg num_classes = 43 for training. I also use the same training, and testing method for GhostNet 1.0 and MobileNetV3 Large. Could you please point out some points in the training code that made my testbench not work with your work? Thank you in advance.

```
import torch
import torchvision
from torchvision import models
import torch.nn as nn
import torch.optim as optim
import torchvision.transforms as transforms
from torch.utils.data import DataLoader
from torchsummary import summary
from utils import save_plots
from levit_c import LeViT_c_128S
mean=(0.485, 0.456, 0.406)
std=(0.229, 0.224, 0.225)
transform_train = transforms.Compose([
    transforms.Resize((224,224)),
    transforms.ToTensor(),
    transforms.Normalize(mean,std),
])
trainset = torchvision.datasets.GTSRB(root='data', download=False, transform=transform_train)   # download=True if you did not download yet
trainloader = DataLoader(trainset, batch_size=32, shuffle=True, num_workers=8) 
model = LeViT_c_128S(num_classes=43)
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model = model.to(device)
# Config Training HyperParameter
criterion = torch.nn.CrossEntropyLoss()
optimizer = optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4)
scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=20, gamma=0.5)
# Lists to keep track of losses and accuracies.
train_loss= []
train_acc= []
# Training
epochs = 50
model.train()
epoch_acc = 0
epoch_loss = 0
for epoch in range(epochs):
    print("\n Epoch: %d"%(epoch+1))
    sum_loss = 0.0
    correct = 0.0
    total =0.0
    for i, data in enumerate(trainloader,0):
        length = len(trainloader)
        inputs,labels = data
        inputs,labels = inputs.to(device),labels.to(device)
        optimizer.zero_grad()
        # forward+backward
        outputs, x = model(inputs)
        loss = criterion(outputs,labels)
        loss.backward()
        optimizer.step()
        # 每个epoch输出损失和正确率
        sum_loss += loss.item()
        _, predicted = torch.max(outputs.data,1)
        total += labels.size(0)
        correct += predicted.eq(labels.data).cpu().sum()
        print("[epoch:%d, iter:%d] Loss: %.03F | Acc: %.3f%%"
              %(epoch+1, (i+1+epoch*length), sum_loss/(i+1), 100.*correct/total))
    scheduler.step()      # Adjust Learning Rate for next epoch
    epoch_loss = sum_loss/(i+1)
    epoch_acc = 100.*correct/total
    train_loss.append(epoch_loss)
    train_acc.append(epoch_acc)
#Display Training Result
model_name = "LeViT_128s"
save_plots(model_name, train_acc, train_loss)
print("Model: LeViT_128s")
print(f"Training Hyperparameter - Epochs: %s, Batch-size: 32, Learning-rate: 0.1, Optimizer: SGD, Momentum: 0.9 " % epochs)
print("[epoch:%d, iter:%d] Loss: %.03F | Acc: %.3f%%" %(epoch+1, (i+1+epoch*length), sum_loss/(i+1), 100.*correct/total))
print(f"Model was saved as %s.pth" % model_name)   
torch.save(model.state_dict(),'LeViT_128s.pth')
```

![acc](https://user-images.githubusercontent.com/66732238/199952559-ba179f24-fd26-4983-aac1-9a47f8399d30.png)
![loss](https://user-images.githubusercontent.com/66732238/199952567-b559d10c-a127-43c0-bb94-63751e0589bf.png)

**Comment by btgraham** [comment]
Is the problem that the training accuracy goes up to 100%, but that it does not generalize to the test set?
LeViT was created with ImageNet in mind, and the statistics of ImageNet images are quite different to GTSRB images.
Also, the data augmentation (no crop augmentation, etc) might be ok for ConvNets, but suboptimal for ViT models.

**Comment by thaihoangminhtam** [comment]
Hi,

Thank you for your reply, I have tried to adjust the training hyperparameters and data augmentation. The LeVit_128S now can achieve 30% accuracy - 346FPS,  and LeViT_192 60% accuracy - FPS 277. While the MobileNetV3 Large can achieve 98.37%-355FPS, GhostNet 1.0 97.26%-275FPS. I use the same training and testing method for 3 models but the LeVit models do not fit my method and I really want to improve. That's why I contact you, I would love to see your model working as expected. :(  The benchmark illustrates your LeViT_128s have higher accuracy and speed than both MobileNet and GhostNet. I'm looking forward to your advice. 
I wonder whether my training method is correct: I download the Pytorch model of MobileNetV3, GhostNet, and your LeVit then load it with config (large for MobileNetV3, width = 1.0 for GhostNet, default for LeVit_128S/192), and num_classes = 43 for all 3 models. Then I use the training source as above. After that, I save the model and then load it in another script for testing.
![Screenshot from 2022-11-04 18-09-48](https://user-images.githubusercontent.com/66732238/200067631-1d51be6f-7fe4-49db-bb8b-8a6ddc1dfedd.png)
(https://colab.research.google.com/drive/1yPpAtsibYw8jkzOarLBhdXnWvJR0Uqlo?usp=sharing#scrollTo=r377ZaaKKImT)

**Comment by btgraham** [comment]
Sorry, but I don't know enough about GTSRB  to help.
